rails/app/models/portal/offering.rb

```diff
@@ -49,7 +49,14 @@ def searchable_attributes
 
   end
 
-  def active?
+  def active?(current_user = nil)
+    active = self.active
+    if current_user
+      metadata = UserOfferingMetadata.find_by(user_id: current_user.id, offering_id: self.id)
+      if metadata.present?
+        active = metadata.active
+      end
+    end
     active
   end
```

Comment:
This is a pre-existing issue and probably not a big deal, but I wonder about all the uses of `active` here. It's used as an attribute name, a method name, and now a local variable within the method. Would it be better to differentiate all those things?